doc/package-notes.xml

```diff
@@ -654,6 +654,68 @@ overrides = self: super: rec {
 </screen>
   </section>
  </section>
+ <section xml:id="sec-firefox">
+  <title>Firefox</title>
+
+  <section xml:id="sec-firefox-config">
+   <title>Configuring the Firefox Wrapper</title>
+
+   <para>
+    The firefox (wrapper) package can be configured through several mechanisms.
+    The older
+    <link xlink:href="https://support.mozilla.org/en-US/kb/customizing-firefox-using-autoconfig">
+    AutoConfig </link> method, and the more recent
+    <link xlink:href="https://github.com/mozilla/policy-templates/blob/master/README.md#extensions">
+    enterprise policies</link> corresponding to the <literal> extraPrefs
+    </literal> and <literal> extraPolicies </literal> arguments of the firefox
+    wrapper package.
+   </para>
+
+   <para>
+    For example the option <literal>media.autoplay.default=1</literal> which
+    disables autoplaying videos, could be specified as follows (the syntax of
+    AutoConfig is a superset of user.js):
+<programlisting>
+firefox.override { extraPrefs = '' pref("media.autoplay.default",1); ''; }
+</programlisting>
+    To disable upload of telemetry data, you could use:
+<programlisting>
+firefox.override { extraPolicies = { DisableTelemetry = true; }; }
+</programlisting>
+    Notice that <literal>extraPolicies</literal> uses nix syntax, which will
+    get converted to json internally for the policies.json file.
+   </para>
+
+   <para>
+    The policies framework can be used to enforce the installation of
+    extensions as follows:
```

Comment:
Before showing how to install extensions, we should list the caveats:

- installs into user dir, no automatic unintallation
- **no upgrades** which will be a dealbreaker to most

Reply:
Sorry, what do you mean by "no upgrades" exactly?
The addon should update as usual, unless you specify an exact version in the url and lock this, I guess.

Reply:
I'm not sure that is true. First of all I would assume (but don't know for sure), that a link like https://addons.mozilla.org/firefox/downloads/file/1189923/noscript_security_suite-latest-fx.xpi links to a specific version (despite having latest in the filename, as there is also a specific fileid).

And even if that isn't the case, there is this: https://github.com/mozilla/policy-templates/issues/207

Have you tried this with an addon that updated afterwards?

Reply:
https://github.com/mozilla/policy-templates/issues/158 explains what the "latest" part in the url does. I did not try with local extensions, I will do that.